sherpa/astro/ui/utils.py

```diff
@@ -9261,27 +9275,92 @@ def fake_pha(self, id, arf, rmf, exposure, backscal=None, areascal=None,
             else:
                 d.ungroup()
 
-        # Update background here.  bkg contains a new background;
-        # delete the old background (if any) and add the new background
-        # to the simulated data set, BEFORE simulating data, and BEFORE
-        # adding scaled background counts to the simulated data.
-        bkg_models = {}
-        if bkg is not None:
-            if bkg == 'model':
-                bkg_models = {1: self.get_bkg_source(id)}
-            else:
-                for bkg_id in d.background_ids:
-                    d.delete_background(bkg_id)
-                self.set_bkg(id, bkg)
+        # If bkg is None then there is no background component, which
+        # means removing any background dataset or models.  They will
+        # need to be added back after the call to fake_pha.
+        #
+        # If bkg="model" then this is already included (as long as
+        # background models are set, but we do not force this
+        # requirement here).
+        #
+        # If bkg is a DataPHA object then remove all the existing
+        # backgrounds and replace them with the background, and set
+        # include_bkg_data. There is also the need to restore a
+        # background model if set: given that there's only the
+        # possibility of using a single background dataset this
+        # potentially loses information, but it's unclear what the
+        # user really expects here given the existing API.
+        #
+        include_bkg_data = False
```

Comment:
I will note that I'm not happy that all the complexity has moved back into this version of fake_pha, as it makes it hard for users of the "OO" interface to replicate this logic. However, there is a reason for this - the handling of background / model data can be rather different between the UI and "OO" interfaces.

Still, I'd hope we can clean this up so that more of the logic is in `sherpa.astro.fake`, but maybe as a later PR.